module/localisation/RobotLocalisation/src/RobotLocalisation.cpp

```diff
@@ -259,10 +255,18 @@ namespace module::localisation {
 
                 if (log_level <= NUClear::DEBUG) {
                     auto state_cell = position_in_map(state, Eigen::Vector2d(0.0, 0.0));
-                    emit(graph("State (x,y)", state_cell.x(), state_cell.y()));
+                    emit(graph("World (x,y)", state_cell.x(), state_cell.y()));
                     // Emit the cell 0.5m in front of the robot to visualise the direction the robot is facing
                     auto direction_cell = position_in_map(state, Eigen::Vector2d(0.5, 0.0));
-                    emit(graph("State (theta)", direction_cell.x(), direction_cell.y()));
+                    emit(graph("World (theta)", direction_cell.x(), direction_cell.y()));
+                    Eigen::Isometry3d Hwc = Hcw.inverse();
+                    Eigen::Vector2d rCWw  = Eigen::Vector2d(Hwc.translation().x(), Hwc.translation().y());
+                    auto robot_cell       = position_in_map(state, rCWw);
+                    emit(graph("Robot (x,y)", robot_cell.x(), robot_cell.y()));
+                    Eigen::Vector3d rPCc  = Eigen::Vector3d(0.5, 0.0, 0.0);
+                    Eigen::Vector3d rPWw  = Hwc * rPCc;
+                    auto robot_theta_cell = position_in_map(state, rPWw.head(2));
+                    emit(graph("Robot (theta)", robot_theta_cell.x(), robot_theta_cell.y()));
```

Comment:
this is a lot of lines for no comments or newlines to break it up

Reply:
I can't comment on it but below here there's code making the message. I assume state is some field/world measurement. can we rename `position` in Field.proto to like Hfw or Hwf whichever it is